controllers/volumereplication_controller.go

```diff
@@ -52,8 +54,29 @@ type VolumeReplicationReconciler struct {
 func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = r.Log.WithValues("volumereplication", req.NamespacedName)
 
-	// your logic here
+	var volumeReplication replicationv1alpha1.VolumeReplication
+	err := r.Client.Get(context.TODO(), req.NamespacedName, &volumeReplication)
+	if err != nil {
+		// reconcile started but object is not found
+	}
 
+	if volumeReplication.Status.State == replicationv1alpha1.ReplicationState("Primary") &&
```

Comment:
this might not be the correct check. please double-check what is the supported state in status. State.

Reply:
Yes this is not implemented.
We do not keep track of `LastKnownImageState`, we only have `State` as `Reconciling` and `Failure`.